### ISSUE TITLE
Airboat no longer functional in HL2:DM

### ISSUE THREAD

**Issue by kisak-valve** [ValveSoftware/Source-1-Games#2440]
Issue transferred from https://github.com/ValveSoftware/source-sdk-2013/issues/440.
@Xeogin posted on 2017-11-22T16:21:35:

Though only commonly used in "puzzle" maps, the airboat is no longer able to turn when attempting to drive it, making it nearly useless.

**Comment by kisak-valve** [comment]
Related to #1702. Marking as a feature request as this does not look like an intended feature.

**Comment by Xeogin** [comment]
This issue can be closed. Was able to fix the functionality by adding the old script file to the custom folder.

**Comment by kisak-valve** [comment]
Closing per the last comment.

**Comment by ID86s** [comment]
this is typo from dev in "wheele_rr" variable (must called "wheel_rr")

**Comment by ID86s** [comment]
> This issue can be closed. Was able to fix the functionality by adding the old script file to the custom folder.

how? can you explain more?

upd - recompiling a map seems fix airboats, but broke other stuff like water, scripts and textures, so that's issue with vehicle scripts because of new engine, there is example

old:
```
    "steering"
    {
        "degrees"        "26"
```
new:
```
   "steering"
    {
        "degreesSlow"        "26"
        "degreesFast"        "26"    
```

**Comment by Xeogin** [comment]
The script I used can be found here: https://github.com/Xeogin/xeogin.github.io/blob/master/hl2mp/custom/coreredux/scripts/airboat.txt

Also using this: https://github.com/Xeogin/xeogin.github.io/blob/master/hl2mp/custom/airboat-wheele_fix.vpk

That said, I rarely run maps using airboats and haven't tested since the Anniversary Update

**Comment by ID86s** [comment]
> The script I used can be found here: https://github.com/Xeogin/xeogin.github.io/blob/master/hl2mp/custom/coreredux/scripts/airboat.txt
> 
> Also using this: https://github.com/Xeogin/xeogin.github.io/blob/master/hl2mp/custom/airboat-wheele_fix.vpk
> 
> That said, I rarely run maps using airboats and haven't tested since the Anniversary Update

it doesn't work for me (maybe i'm doing something wrong, i need to place those files in hl2dm/custom right?), tested on pre and post anniversary update :(
old maps like: js_build_puzzle_nine_v14, js_build_puzzle_fifteen_v5 and js_coop_airboats_v5 still broken (gamepad tested too)

Error:
`DataTable warning: (class prop_vehicle_airboat): Out-of-range value (-0.019378 / 0.000000) in SendPropFloat 'm_flCycle', clamping.`

but updated maps like: js_build_puzzle9_ob_v2 and js_puzzle_clare_reloaded_v6 works fine

**Comment by Xeogin** [comment]
> Replying to https://github.com/ValveSoftware/Source-1-Games/issues/2440#issuecomment-2595231312

You are correct, thank you for supplying example maps. It turns out the older maps are broken because they have the script file baked in: https://forums.sourceop.com/threads/3484-SOLVED-Bugs-source-op?s=dbc99a45d1b4bb36601e515aed5786ba&p=62349&viewfull=1#post62349

As a workaround, using something like Stripper Source to change the vehicle script used on load resolves the issue. You may have noticed my script is in the wrong location since the usual one is in /scripts/vehicles/airboat.txt. This serves to help facilitate the change as I have a global filter that replaces the entries with /scripts/airboat.txt

**Comment by ID86s** [comment]
> Replying to https://github.com/ValveSoftware/Source-1-Games/issues/2440#issuecomment-2606428652

thanks, i just stripped airboat.txt from map and replaced entire "steering" section (some maps are use custom airboat parameters like in js_coop_airboats_v5) in it and it works!
